File: py/job.py
from typing import List

class Job:
    def __init__(self, id: str, deadline: int, profit: int):
        self.id = id
        self.deadline = deadline
        self.profit = profit

def schedule_jobs(jobs: List[Job]):
    jobs.sort(key=lambda x: x.profit, reverse=True)

    n = len(jobs)
    slot = [False] * n
    result = [''] * n

    for i in range(n):
        for j in range(min(n, jobs[i].deadline) - 1, -1, -1):
            if not slot[j]:
                result[j] = jobs[i].id
                slot[j] = True
                break

    print("Scheduled Jobs:", " ".join(result))

if __name__ == "__main__":
    jobs = [
        Job('a', 2, 100),
        Job('b', 1, 19),
        Job('c', 2, 27),
        Job('d', 1, 25),
        Job('e', 3, 15)
    ]

    schedule_jobs(jobs)

File: job.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

struct Job {
    char id;
    int deadline;
    int profit;

    Job(char id, int deadline, int profit) : id(id), deadline(deadline), profit(profit) {}
};

bool compare(Job& j1, Job& j2) {
    return j1.profit > j2.profit;
}

void scheduleJobs(vector<Job>& jobs) {
    sort(jobs.begin(), jobs.end(), compare);

    int n = jobs.size();
    vector<bool> slot(n, false);
    vector<char> result(n);

    for (int i = 0; i < n; i++) {
        for (int j = min(n, jobs[i].deadline) - 1; j >= 0; j--) {
            if (!slot[j]) {
                result[j] = jobs[i].id;
                slot[j] = true;
                break;
            }
        }
    }

    cout << "Scheduled Jobs: ";
    for (char jobId : result) {
        if (jobId != '\0') {
            cout << jobId << " ";
        }
    }
    cout << endl;
}

int main() {
    vector<Job> jobs = {
        Job('a', 2, 100),
        Job('b', 1, 19),
        Job('c', 2, 27),
        Job('d', 1, 25),
        Job('e', 3, 15)
    };

    scheduleJobs(jobs);

    return 0;
}
